fileutil/fileutil.go

```diff
@@ -23,3 +24,25 @@ func ReadDir(dirpath string) ([]string, error) {
 	sort.Strings(names)
 	return names, nil
 }
+
+// Rename safely renames a file.
```

Comment:
This has a race condition between the remove and the rename

Reply:
The doc says it's "safe" however there's a race condition. In what way is the meant to be safe?

Reply:
I meant to elaborate on what exactly the race condition is.

Reply:
From the time the removeall starts to the rename ends, partial or no file may be present.